lib/model.js

```diff
@@ -214,13 +214,6 @@ _.assign(props, {
   mongodb_database_name: {
     type: 'string',
     default: undefined
-  },
-  /**
-   * Whether BSON values should be promoted to their JS type counterparts.
-   */
-  promote_values: {
-    type: 'boolean',
-    default: true
```

Comment:
setting it to `undefined` (or just not setting a default at all) should work. We are relying on that functionality in many places, for example in the [Compass preferences model](https://github.com/10gen/compass/blob/master/src/app/models/preferences.js#L30-L33).

Ampersand only disallows this if the `required: true` option is set. https://ampersandjs.com/docs/#ampersand-state-props

`driver_options` is a derived property and should not be modified directly, or it gets out of sync with its dependent props. I was hoping it is disallowed even, but apparently it can be modified because it is an object. Only if you try to replace it does Ampersand complain: 

```
c.driver_options = {}
TypeError: `driver_options` is a derived property, it can't be set directly.
```


We can talk in person if it helps.